content/2021-11-05-introducing-drogue-server/index.md

```diff
@@ -0,0 +1,123 @@
+
+
+# Installing The Binary
+
+One of the goals with running Drogue Cloud locally was to encapsulate the functionality in a single binary, `drogue-cloud-server`. You can [build it yourself](https://github.com/drogue-iot/drogue-cloud/tree/main/server#building), or you can [download prebuilt binaries]() for your platform.
+
+# Running
+
+Once installed, it's time to run. You can find the detailed instructions in [the book](https://book.drogue.io/drogue-cloud/dev/deployment/single-binary.html). It's important to note that for the time being, the Kafka, Keycloak and PostgreSQL instance must be reachable on localhost where the server is running.
+
+Once you have the dependencies running, we can start the server:
+
+```rust
+./drogue-cloud-server run --enable-all
+```
+
+When starting, the server will perform the following steps (in order):
+
+* Connect to PostgreSQL and run schema migration
+* Connect to Keycloak and create an OIDC client
+* Launch the services specified for the run command
+
+Using `--enable-all` will run all Drogue Cloud services supported. At the time of writing, this is:
+
+* REST API
+* Device registry
+* Device authentication
+* User authentication
+* HTTP endpoint
+* MQTT endpoint
+
+We plan to expose all the Drogue Cloud services.
+
+The server will print some info to the terminal on how to log into the server using the [drg](https://github.com/drogue-iot/drg) client, creating applications and devices, and publishing telemtry data:
+
+```bash
+$ drogue-cloud-server run --enable-all
+Migrating database schema...
+Migrating database schema... done!
+Configuring keycloak... done!
+Drogue Cloud is running!
+
+Endpoints:
+        API:     http://localhost:10001
+        HTTP:    http://localhost:8088
+        MQTT:    mqtt://localhost:1883
+
+Keycloak Credentials:
+        User: admin
+        Password: admin123456
+
+Logging in:
+        drg login http://localhost:10001
+
+Creating an application:
+        drg create app example-app
+
+Creating a device:
+        drg create device --app example-app device1 --spec '{"credentials":{"credentials":[{"pass":"hey-rodney"}]}}'
+
+Publishing data to the HTTP endpoint:
+        curl -u 'device1@example-app:hey-rodney' -d '{"temp": 42}' -v -H "Content-Type: application/json" -X POST http://localhost:8088/v1/foo
+```
+
+## Other options
+
+The server can also be run with other options, we'll quickly cover the most important:
+
+* `--server-cert` and `--server-key` - enable TLS for the endpoints. Both arguments should refer to PEM-encoded files.
```

Comment:
I don't know how complicated would be to provide an example of creating these and using TLS?

Reply:
Yeah, I'll add that. Thanks!